include/modem/location.h

```diff
@@ -0,0 +1,299 @@
+	 *
+	 * @details Index 0 has the highest priority. Number of methods is indicated in
+	 * 'methods_count'.
+	 */
+	struct loc_method_config *methods;
+	/**
+	 * @brief Position update interval in seconds.
+	 *
+	 * @details Set to 0 for a single position update. For periodic position updates
+	 * the valid range is 10...65535 seconds.
+	 */
+	uint16_t interval;
+};
+
+/**
+ * @brief Event handler prototype.
+ *
+ * @param[in] event_data Event data.
+ */
+typedef void (*location_event_handler_t)(const struct loc_event_data *event_data);
+
+/** @brief Initializes the library.
+ *
+ * @details Initializes the library and sets the event handler function.
+ *
+ * @param[in] event_handler Event handler function.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_init(location_event_handler_t event_handler);
+
+/**
+ * @brief Requests the current position or starts periodic position updates.
+ *
+ * @details Requests the current position using the given configuration. Depending on the
+ *          configuration, a single position or periodic position updates are given. The results are
+ *          delivered to the event handler function.
+ *
+ *          Periodic position updates can be stopped by calling location_cancel().
+ *
+ * @param[in] config Used configuration or NULL to get a single position update with
+ *                   the default configuration.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_request(const struct loc_config *config);
+
+/**
+ * @brief Cancels periodic position updates.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int location_request_cancel(void);
+
+/**
+ * @brief Sets default values to given configuration.
+ *
+ * @details Given methods are set as list of methods for the given configuration.
+ * Caller is responsible for allocating and deallocating memory for both configuration and
+ * method list. 'methods' memory area must be kept allocated as long as the configuration is used.
+ *
+ * @param[in,out] config Configuration which is supplied with default values.
+ * @param[in] methods_count Number of methods to be stored in 'methods'.
+ * @param[in] methods List of methods. This will be initialized to zero.
+ */
+void loc_config_defaults_set(struct loc_config *config, uint8_t methods_count,
+			     struct loc_method_config *methods);
+
+/**
+ * @brief Sets default values for given method configuration based on given type.
+ *
+ * @details Intended use is that this is part of loc_config that has just been initialized, i.e.,
+ * this function is called right after loc_config_defaults_set().
+ *
+ * @param[in,out] method Method configuration which is supplied with default values.
+ * @param[in] method_type Location method type.
+ */
+void loc_config_method_defaults_set(struct loc_method_config *method, enum loc_method method_type);
+
+#if defined(CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL)
```

Comment:
Good question. I don't have a strong argument either way. Do you have an opinion @trantanen ?

Reply:
@rlubos Do we have a practice on whether to flag function declarations inside configuration flags, or have an almost empty function that would return `-ENOTSUP`?

Reply:
I don't think we have a written rule for that, but it's a common practice for instance in networking code to provide an empty stub function in case a feature is not enabled, for instance:
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/net/net_if.h#L1428
Or optionally, it's also acceptable to leave a declaration, and #ifdef the function body.

Reply:
I'll pick the option "leave a declaration, and #ifdef the function body."